inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -290,7 +290,7 @@ def extract_authors_from_xml(xml_content):
 
         # Getting all the author ids
         for source, id in itertools.izip(author.xpath('./authorIDs/authorID[@source!="" and text()!=""]/@source | ./authorids/authorid[@source!="" and text()!=""]/@source').getall(), author.xpath('./authorIDs/authorID[@source!="" and text()!=""]/text() | ./authorids/authorid[@source!="" and text()!=""]/text()').getall()):
-            if not re.match("undefined", source, flags=re.IGNORECASE) or not re.match("undefined|inspire-", id, flags=re.IGNORECASE):
+            if not re.match("undefined", source, flags=re.IGNORECASE) and not re.match("undefined|inspire-$", id, flags=re.IGNORECASE):
```

Comment:
I would check also if there are no whitespaces after (maybe it would be parsed as something like `INSPIRE- `) and your regex will not match it